proposed/2023/sponsor-link.md

```diff
@@ -0,0 +1,164 @@
+    </sponsors>
+</package>
+```
+
+For .NET projects that use the `PackageReference` format which draws inputs from the project file, one can create a new `<ItemGroup>` for their sponsors such as:
+
+```xml
+<ItemGroup>
+    <Sponsors Include="Patreon" Url="https://www.patreon.com/mypatreonusername"/>
+    <Sponsors Include="GitHub Sponsors" Url="https://github.com/sponsors/mygithubusername"/>
+    <Sponsors Include="Tidelift" Url="https://tidelift.com/subscription/pkg/mypackagename"/>
+</ItemGroup>
+```
+
+#### NuGet.org
+
+On NuGet.org there will be a new `Sponsor this package` button which will open up a webpage to the first defined sponsor in the package metadata. Given there can be multiple sponsors, there may be an additional experience where one can click "all sponsors" and un-collapse the other options available.
+
+![NuGet.org Sponsor Experience](../2023/resources/NuGetOrgSponsors.png)
+
+#### Visual Studio
+
+In Visual Studio's NuGet package manager UI, there will be a new link that says `Sponsor` which will open up a webpage to the first defined sponsor in the package metadata.
+
+![Visual Studio Sponsor Experience](../2023/resources/VisualStudioSponsors.png)
+
+#### dotnet restore
+
+If a developer has the MSBuild property of `<NuGetSponsor>true</NuGetSponsor>` they will see a summary of packages that require funding and an empowering message on how to see the funding needs after a restore.
+
+A simple message will appear at the bottom such as:
+
+```
+X packages are looking for sponsors. Run `dotnet list package --sponsor` for more details.
+```
+
+#### dotnet list package --sponsor
+
+To list all the packages that are looking for sponsors, one can run `dotnet list package --sponsor` which will provide output for each project. If no sponsors are listed, the project will provide a message saying there are no packages looking for sponsors. Finally, an empowering message will be provided regarding the follow-up command to learn more about sponsoring an individual package:
+
+```plaintext
+Project 'Umbraco.Core' has the following package references looking for sponsors
+   [net7.0]: 
+   Top-level Package                                              Requested            Resolved      
+   > Moq                                                          4.18.2               4.18.2
+   > Serilog                                                 	  2.12.0               2.12.0
+   
+The given project `Umbraco.Web.UI` has no packages looking for sponsors given the current sources.
+
+To learn more about sponsoring a package, run `dotnet sponsor package <PACKAGE_NAME>`.
+```
+
+#### dotnet sponsor package <PACKAGE_NAME>
+
+When an individual is feeling generous to sponsor a package, they can run the `dotnet sponsor package <PACKAGE_NAME>` command which will open up the default browser on their machine with the url of the first sponsor defined in the package metadata.
```

Comment:
I am a little squeamish about creating a new top-level verb in the CLI for this, but there are no commands contributed by .NET SDK Tools on NuGet.org that I can see that would be squatting the verb, and other ecosystems also have it as a top-level.  I'd be interested in thoughts about other commands the `sponsor` verb might grow in the future as a thought exercise.

Reply:
Why not `dotnet nuget sponsor`?

Reply:
In the CLI we don't generally make as much of a distinction about which piece of tech is contributing the functionality - we try to focus on the action the user is taking, then do whatever transformations are required to make that happen.  An example of this is `dotnet add package` - the user wants to add a package to a project, they don't necessarily care if it's `nuget.exe` doing it or the CLI itself, etc.  Same theory here - the _action_ is the important thing, not the specific tool that's used to achieve the action.

Reply:
Please, don't make software too "smart", just print a set of links. By saving one click you make it impossible to use it in a script and view other options than the first one. `dotnet add` doesn't randomly download the first package from nuget.org :)

Reply:
@WhiteBlackGoose It's a good point. This behavior was inspired by what npm does and I talked to their PM on this and what their rationale was for default behavior. Agreed that providing users choice for how they want to donate is ideal. But we also want to make it frictionless to suggest their preferred way. Hopefully that helps clear up the rationale here.